lib/middleware/request-handler.ts

```diff
@@ -290,6 +239,196 @@ export function onSubscribe(req: Interface.IOurRequest,
+        return next(new restify.UnsupportedMediaTypeError('Unsupported Content-Type.'));
+    }
+
+    return next();
+}
+
+export function elevateRequest (req: Interface.IOurRequest,
+                                res: Interface.IOurResponse,
+                                next: restify.Next): void
+{
+    var chunkIndex: number = 0;
+
+    /**
+     * This is called each time data is added to the response.
+     * @param isEnd Whether this is the last time this will be called for this http request.
+     * @returns {string} Text that should be added to the response before the caller adds its data.
+     */
+    function prepareResponseText(isEnd: boolean): string {
+        var text: string;
+        if (1 === ++chunkIndex) {
+            // This is the first time we're adding response text. Set the headers as well.
+            res.statusCode = 200;
+            res.setHeader('content-type', 'application/json');
+            // If isEnd is true, then we aren't actually adding a data array to the http response.
+            text = isEnd ? '{' : '{"data":[';
+        } else {
+            // If isEnd is true, we must have previously added data to the response, so we need to
+            // close the data array.
+            text = isEnd ? '],' : ',';
+        }
+        assert(text, 'Response text was not set');
+        return text;
+    }
+
+    function stringifyPair ( key: string, value: any ): string {
+        return '"' + key.replace(/'/g, '\\$&') + '":' + JSON.stringify(value);
+    }
+
+    // Right now we only use restify.InternalError(500)
+    // TODO: Add error type differentiation
+    function createError(err: Error): any {
+        return new restify.InternalError(err.message);
+    }
+
+    res.sendChunk = (data: any): void => {
+        var text = prepareResponseText(false);
+        res.write(text + JSON.stringify(data));
+    };
+
+    res.sendEnd = (status: string, message: string): void => {
+        var text = util.format('%s%s,%s}',
+                               prepareResponseText(true),
+                               stringifyPair('status', status),
+                               stringifyPair('message', message || ''));
+        res.end(text);
+    };
+
+    res.sendError = (err: Error): void|any => {
+        // send Error if this is the first chunk
+        if (chunkIndex === 0) {
+            return createError(err);
+        }
+        // Otherwise, set -1 as status code and error message
+        return res.sendEnd(-1, err.message);
+    };
+
+    return next();
+}
+
+export function onRequest(req: Interface.IOurRequest,
+                          res: Interface.IOurResponse,
+                          next: restify.Next): void
+{
+    assert(req.blpSession, 'blpSession not found');
+
+    var uri = util.format('//%s/%s', req.query.ns, req.query.service);
+    var callback = getRequestCallback(req, res, next);
+    req.blpSession.request(uri, req.query.type, req.body, req.identity, callback);
```

Comment:
I understand what is happening here, but it seems misleading having this function take `next` without having it visible that it gets called.  I would advocate for calling next explicitly here or reworking the function signature and how everything fits together.